src/reducers/ast.js

```diff
@@ -187,12 +187,12 @@ export function isEmptyLineInSource(
   return emptyLines && emptyLines.includes(line);
 }
 
-export function getEmptyLines(state: OuterState, source: Source) {
-  if (!source) {
+export function getEmptyLines(state: OuterState, sourceRecord: SourceRecord) {
+  if (!sourceRecord) {
     return null;
   }
 
-  return state.ast.emptyLines.get(source.id);
+  return state.ast.emptyLines.get(sourceRecord.id);
```

Comment:
Maybe instead of the full source record just pass in the `id` to `getEmptyLines`? Then we could just return if `!id`. Question if there are any edge cases where there is a source without an id that would break this approach...

Reply:
good idea, lets do that